src/newgrf_gui.cpp

```diff
@@ -1464,6 +1464,7 @@ struct NewGRFWindow : public Window, NewGRFScanCallback {
 				this->avails.push_back(c);
 			} else {
 				const GRFConfig *best = FindGRFConfig(c->ident.grfid, HasBit(c->flags, GCF_INVALID) ? FGCM_NEWEST : FGCM_NEWEST_VALID);
+				assert(best != nullptr); // At worst case the GRF of this loop iteration is found
```

Comment:
I fail to decipher what the comment means?

Reply:
Is `Best will never be nullptr, as it will either find a newer NewGRF or it will find itself, i.e. best == c.` better?
Yeah, in hind sight it's not a really good comment. No idea why exactly anymore, although "itself" of the proposal is not really clear on its own as it's not super obvious whether that refers to `best` or `c`. That might be why it was mentioning the loop iteration, so it points to `c` but that didn't make it much clearer.

Reply:
Ah, so, you mean to say: will never trigger. Maybe that is the valuable piece of information here. Suggestion:

`Never triggers; FindGRFConfig returns either c, or a newer version of c`

Or, as extension to your suggestion:

`Cannot trigger: best will never be nullptr, as it will either find a newer NewGRF or it will find itself, i.e. best == c`

But yeah, even without the "cannot trigger" this comment is a lot more clear to me :) Go for it!